src/v2/services/implementations/AssetsService.ts

```diff
@@ -28,35 +29,35 @@ export class AssetsService implements IAssetsService {
   }
 
   public async transferNativeAsset(param: ParamAssetTransfer): Promise<void> {
+    const { decimal } = useChainMetadata();
     const useableBalance = await this.AssetsRepository.getNativeBalance(param.senderAddress);
-    const isBalanceNotEnough =
-      Number(ethers.utils.formatUnits(useableBalance, 18)) -
-        Number(ethers.utils.formatUnits(param.amount, 18)) <
+    const isBalanceEnough =
+      Number(ethers.utils.formatUnits(useableBalance, decimal.value)) -
+        Number(ethers.utils.formatUnits(param.amount, decimal.value)) >
```

Comment:
Why don't just use `ethers.utils.formatEther(wei)`? So you don't need to use `useChainMetadata`.